app/assets/javascripts/components/course.jsx

```diff
@@ -46,7 +47,8 @@ const Course = createReactClass({
     ServerActions.fetch('course', courseID);
     this.props.fetchUsers(courseID);
     ServerActions.fetch('users', courseID);
-    return ServerActions.fetch('campaigns', courseID);
+    this.props.fetchAllCampaigns();
```

Comment:
This will fetch all campaigns every time the page is loaded (whether or not the user is even able to add/remove campaigns). Instead of doing that here, I think this should be done closer to when/where it's needed... for example, when the user first focuses the campaign button input.